MIGRATION.MD

```diff
@@ -0,0 +1,234 @@
+# Migration from 1.x to 2.0
+
+The main changes introduced in SDK 2.0 compared to 1.x are:
+
+1. All the products (RUM, Trace, Logs, etc.) are now extracted into different modules. That allows you to integrate only what is needed into your application.
+
+In comparison with version 1.x where all products were in a single artifact `com.datadoghq:dd-sdk-android:x.x.x`, now you need to adopt the following artifacts instead:
+
+* RUM: `com.datadoghq:dd-sdk-android-rum:x.x.x`
+* Logs: `com.datadoghq:dd-sdk-android-logs:x.x.x`
+* Trace: `com.datadoghq:dd-sdk-android-trace:x.x.x`
+* WebView Tracking: `com.datadoghq:dd-sdk-android-webview:x.x.x`
+
+**NOTE**: In case of NDK Crash Reporting and WebView Tracking usage it's also needed to add RUM and/or Logs artifacts to be able to report events to RUM and/or Logs respectively.
+
+Reference to the `com.datadoghq:dd-sdk-android` artifact should be removed from your Gradle buildscript, this artifact doesn't exist anymore.
+
+All other artifacts coordinates stay the same.
+
+2. Support for multiple SDK instances.
+
+Previously Datadog SDK implemented a singleton and only one SDK instance could exist in the application process. This created obstacles for use-cases like the usage of the SDK by 3rd party libraries.
+
+With version 2.0 we addressed this limitation:
+
+* Now it is possible to initialize multiple instances of the SDK, associating them with a name.
+* Many methods of the SDK can optionally take a SDK instance as an argument. If not provided, the call will be associated with the default (nameless) SDK instance.
+
+Here is an example illustrating the typical usage in case of default instance vs named instance:
+
+```kotlin
+// default instance
+Datadog.initialize(context, configuration, trackingConsent)
+val userInfo = UserInfo(...)
+Datadog.setUserInfo(userInfo)
+
+// named instance
+val namedSdkInstance = Datadog.initialize("myInstance", context, configuration, trackingConsent)
+val userInfo = UserInfo(...)
+Datadog.setUserInfo(userInfo, sdkCore = namedSdkInstance)
```

Comment:
Is it? it should be using the core registry, no?

```kotlin
Datadog.initialize("myInstance", context, configuration, trackingConsent)
val userInfo = UserInfo(...)
Datadog.setUserInfo(userInfo, sdkCore = Datadog.sdkInstance("myInstance"))
```

Reply:
this can be done this way as well, but on Android `Datadog.initialize` returns SDK instance, so we can use it directly.